common/src/main/java/org/apache/uniffle/common/netty/client/RpcResponseCallback.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.common.netty.client;
+
+import org.apache.uniffle.common.netty.protocol.RpcResponse;
+
+public interface RpcResponseCallback {
+  /**
+   * Successful serialized result from server.
+   *
+   * <p>After `onSuccess` returns, `response` will be recycled and its content will become invalid.
+   * Please copy the content of `response` if you want to use it after `onSuccess` returns.
+   */
+  void onSuccess(RpcResponse rpcResponse);
```

Comment:
In fact, not all response types are RPCResponse, such as getInMemoryShuffleData.

Reply:
getInMemoryShuffleData result will extends RPCResponse in your design?

Reply:
`GetMemoryShuffleDataResponse` will extend `Message`.

Reply:
Why do `GetMemoryShuffleDataResponse` need extend `Message`?

Reply:
Seems every Response type will have `status` and `retMsg` attr, it can extend RPCResponse, my mistake.